### PR TITLE
Feat: Expose the gateway name as a macro

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Review by GeorgeSittas — APPROVED** [review]

File: sqlmesh/core/config/root.py [hunk 1]
```diff
@@ -152,6 +152,14 @@ def get_scheduler(self, gateway_name: t.Optional[str] = None) -> SchedulerConfig
     def get_state_schema(self, gateway_name: t.Optional[str] = None) -> t.Optional[str]:
         return self.get_gateway(gateway_name).state_schema
 
+    @property
+    def default_gateway_name(self) -> str:
+        if self.default_gateway:
+            return self.default_gateway
+        if "" in self.gateways:
+            return ""
+        return next(iter(self.gateways))
```
Comment (tobymao) [inline sqlmesh/core/config/root.py:161]: i think i have a helper for this, sqlglot.helpers#first

File: sqlmesh/utils/metaprogramming.py [hunk 2]
```diff
@@ -304,7 +304,7 @@ def __lt__(self, other: t.Any) -> bool:
 
 
 class Executable(PydanticModel):
-    payload: t.Any
+    payload: str
```
Comment (tobymao) [inline sqlmesh/utils/metaprogramming.py:307]: hm i wonder why this was an any before?

File: sqlmesh/core/context.py [hunk 3]
```diff
@@ -388,6 +388,7 @@ def load(self, update_schemas: bool = True) -> Context:
             gc.disable()
             project = self._loader.load(self, update_schemas)
             self._macros = project.macros
+            self._jinja_macros = project.jinja_macros
```
Comment (tobymao) [inline sqlmesh/core/context.py:391]: was this just oversight? missing?

File: sqlmesh/utils/metaprogramming.py (hunk 2 above)
Comment (izeigerman) [reply sqlmesh/utils/metaprogramming.py:307]: perhaps some artifact from before. As far as I can tell we always pass string in there

File: sqlmesh/core/context.py (hunk 3 above)
Comment (izeigerman) [reply sqlmesh/core/context.py:391]: it was not used anywhere before

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/loader.py [hunk 4]
```diff
@@ -256,6 +257,10 @@ def _load_scripts(self) -> t.Tuple[MacroRegistry, JinjaMacroRegistry]:
         macros = macro.get_registry()
         macro.set_registry(standard_macros)
 
+        gateway_name = self._context.gateway or self._context.config.default_gateway_name
+        macros["gateway"] = Executable.value(gateway_name)
+        jinja_macros.add_globals({"gateway": gateway_name})
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply sqlmesh/core/loader.py:262]: I'd rather wait for at least one more use case before introducing helpers.

[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlmesh/core/loader.py:262]: Sounds good

**Pull request merged**